src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String INDEX_ERROR = "Index is wrong";
+    private int listSize = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private void connection(Node<T> node) {
+        if (head == null) {
+            head = tail = node;
+        } else {
+            tail.next = node;
+            tail = node;
+        }
+    }
+
+    private Node<T> stepBystep(int index) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < index; i++) {
+            currentNode = currentNode.next;
+        }
+        return currentNode;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        connection(newNode);
+        listSize++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > listSize || index < 0) {
```

Comment:
Try to minimize code duplication. For instance, the check for valid index and non-empty list is repeated multiple times. You could create a method `checkIndex` or `validateIndex` to centralize this validation.